harvester/harvester
# Add http-proxy setting

**Problem:**
Unable to use HTTP proxy to access external services, including downloading images and backup to s3 services.

**Solution:**
When the setting is updated, inject HTTP proxy envvars to system pods and the backup secret

**Related Issue:**
Address https://github.com/harvester/harvester/issues/1012
Partially address https://github.com/harvester/harvester/issues/1218, the missing part is envvar propagation to RKE2

**Test plan:**
1. Set up Harvester in a private network environment that can only access the internet via an HTTP proxy
2. Configure the HTTP proxy in the Harvester settings page(minimum NO_PROXY: `localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,.svc,.cluster.local,<your-internal-network>`)
3. Verify creating images from URL and backup to s3 works.
4. Verify updating/removing the proxy configuration works.
